src/main/java/io/anserini/collection/C4Collection.java

```diff
@@ -77,19 +77,24 @@ private int getFileNumber(String fileName) {
     return segments.stream().filter(x -> getFileNumber(x.toString()) % shardCount == currShard).collect(Collectors.toList());
   }
 
-  public static class Segment extends FileSegment<C4Collection.Document>{
-    private MappingIterator<JsonNode> iterator; // iterator for JSON line objects
-    private JsonNode node = null;
-    private String filePath;
-    private String fileName;
-    private int count = 0;
+  public class Segment extends FileSegment<C4Collection.Document>{
+    protected MappingIterator<JsonNode> iterator; // iterator for JSON line objects
+    protected JsonNode node = null;
+    protected String filePath;
+    protected String fileName;
+    protected int count = 0;
 
     public Segment(Path path) throws IOException {
       super(path);
       filePath = path.toString();
       int fileNumStart = filePath.indexOf("c4-train.") + 9;
-      // plus one to remove leading zero
-      fileName = filePath.substring(fileNumStart + 1, fileNumStart + 5);
+      if(C4Collection.this instanceof C4NoCleanCollection){
+        // plus one to remove leading zero
+        fileName = filePath.substring(fileNumStart + 1, fileNumStart + 14);
```

Comment:
this can be much cleaner with a regular expression...

Reply:
Done too :)